site/content/protocols/mediator-coordination/2.0/readme.md

```diff
@@ -0,0 +1,229 @@
+
+First, the `recipient` sends a `mediate-request` message to the `mediator`. If the `mediator` is willing to route messages, it will respond with a `mediate-grant` message, otherwise with a `mediate-deny` message. The `recipient` will share the routing information in the grant message with other contacts.
+
+When a new key is used by the `recipient`, it must be registered with the `mediator` to enable route identification. This is done with a `keylist-update` message.
+
+The `keylist-update` and `keylist-query` methods are used over time to identify and remove keys that are no longer in use by the `recipient`.
+
+## Design By Contract
+
+No protocol specific errors exist. Any errors related to headers or other core features are documented in the appropriate places.
+
+## Security
+
+This protocol expects messages to be encrypted during transmission, and repudiable. 
+
+## Composition
+
+Supported Goal Code | Notes
+--- | ---
+                     |       
+                     |       
+
+
+
+## Message Reference
+
+### Mediate Request
+This message serves as a request from the `recipient` to the `mediator`, asking for the permission (and routing information) to publish the endpoint as a mediator.
+
+Message Type URI: `https://didcomm.org/coordinate-mediation/2.0/mediate-request`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/coordinate-mediation/2.0/mediate-request",
+    }
+}
+```
+
+### Mediate Deny
+This message serves as notification of the `mediator` denying the `recipient`'s request for mediation.
+
+Message Type URI: `https://didcomm.org/coordinate-mediation/2.0/mediate-deny`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/coordinate-mediation/2.0/mediate-deny",
+    }
+}
+```
+
+### Mediate Grant
+A mediate grant message is a signal from the `mediator` to the `recipient` that permission is given to distribute the included information as an inbound route.
+
+Message Type URI: `https://didcomm.org/coordinate-mediation/2.0/mediate-grant`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/coordinate-mediation/2.0/mediate-grant",
+    "body": 
+            {
+                "endpoint": "https://mediators-r-us.com",
+                "routing_keys": ["did:peer:z6Mkfriq1MqLBoPWecGoDLjguo1sB9brj6wT3qZ5BxkKpuP6"]
+            }
+}
+```
+where:
+
+- `endpoint`: the endpoint reported to mediation client connections.
+
+- `routing_keys`: list of keys in intended routing order. Key used as recipient of forward messages.
+
+### Keylist Update
+Used to notify the `mediator` of keys in use by the `recipient`.
+
+Message Type URI: `https://didcomm.org/coordinate-mediation/2.0/keylist-update`
+
+```json
+{
+    "id": "123456780",
+    "type": "https://didcomm.org/coordinate-mediation/2.0/keylist-update",
```

Comment:
should this be `did-list-update`? That's a terrible name, but to reflect we're not dealing with keys anymore, but with dids. Or maybe `recipient-dids-update`?

Reply:
we should change both `keylist` and `keylist-update`. I think that adding the word `recipient` is more descriptive. `recipient-dids` and `recipient-did-update` may work.